Throw validation error after authorization

licitdev opened the pull request:
## Description

<!--

A summary of the changes and a reference to the Issue that was fixed / implemented.

NOTE: All Pull Requests require a corresponding open Issue.

Please reference the Issue number below:

-->

Throw the validation error after authorization but before the mutation occurs, if any.

Fixes ENG-605

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [x] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

licitdev committed:
Throw validation error after authorization

licitdev committed:
Improve tests

paescuj reviewed (changes requested):
Looks very good to me! Just wondering whether we should apply the same logic to `deleteMany` in `users` and maybe even other methods like `requestPasswordReset`? https://github.com/directus/directus/blob/ea1bc45fff2454d06aee32898cd862c4a2f52e01/api/src/services/users.ts#L283

licitdev committed:
Merge branch 'main' into fix/errors-before-authorization

licitdev committed:
Refactor for deleteMany
Co-authored-by: Pascal Jufer <pascal-jufer@bluewin.ch>

licitdev committed:
Refactor other usages in users service that errors before ForbiddenException

licitdev committed:
Update tests

licitdev commented:
@paescuj Great catch, thank you! Updated for `deleteMany`. 👍 
Also refactored for other usages for a consistent behaviour, for the whitelist URLs. 😄

paescuj committed:
Adjust test for 'deleteByQuery'

paescuj reviewed (approved):
Great work!

rijkvanzanten committed:
Merge branch 'main' into fix/errors-before-authorization
